src/test/java/org/apache/sysds/test/AutomatedTestBase.java

```diff
@@ -1283,44 +1288,89 @@ public static int getRandomAvailablePort() {
 			return availableSocket.getLocalPort();
 		}
 		catch(IOException e) {
-			// If no port was found just use 9999
+			// If no port was found just use 9990
 			return 9990;
 		}
 	}
 
+	private static void waitForFederatedWorker(int port) {
+		long start = System.currentTimeMillis();
+		while(System.currentTimeMillis() - start <= FED_WORKER_CONNECTION_WAIT) {
+			FederatedResponse response = null;
+			try {
+				response = executeFederatedOperation(new InetSocketAddress("localhost", port),
+					new FederatedRequest(FederatedRequest.RequestType.CLEAR)).get();
+			}
+			catch(Exception e) {
+				// failed, try again if fed worker wait has not passed yet
+			}
+			if(response != null) {
+				if(!(response.isSuccessful()))
+					LOG.error("Ping failed: " + response.getErrorMessage());
+				else
+					break;
+			}
+		}
+	}
```

Comment:
If the worker does not respond and the time limit has been reached then the execution would just continue as usual from here. This means that it will fail later when it tries to send a request to the worker. This will make it harder to debug compared to handling it when trying to create the worker.